core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/ListCodec.java

```diff
@@ -115,11 +115,19 @@ public ByteBuffer encode(
       int size = input.getInt();
       List<ElementT> result = new ArrayList<>(size);
       for (int i = 0; i < size; i++) {
+        ElementT element;
         int elementSize = input.getInt();
-        ByteBuffer encodedElement = input.slice();
-        encodedElement.limit(elementSize);
-        input.position(input.position() + elementSize);
-        result.add(elementCodec.decode(encodedElement, protocolVersion));
+        // Allow null elements on the decode path, because Cassandra might return such collections
+        // for some computed values in the future -- e.g. SELECT ttl(some_collection)
```

Comment:
I'm probably a bit paranoid but should we include the example of `ttl(collection)`? This example is DSE-specific.

Reply:
[CASSANDRA-8877](https://issues.apache.org/jira/browse/CASSANDRA-8877) will likely add that feature.